src/mavsdk/core/locked_queue.h

```diff
@@ -13,13 +13,13 @@ template<class T> class LockedQueue {
 
     void push_back(std::shared_ptr<T> item_ptr)
     {
-        std::lock_guard<std::mutex> lock(_mutex);
+        std::lock_guard<std::recursive_mutex> lock(_mutex);
```

Comment:
I tried to avoid having to use the recursive mutex. Instead, the user callback needs to be called from the user callback thread pool, e.g.

https://github.com/mavlink/MAVSDK/blob/de90b234f6a8a7ab8aa75e3943de65f2bae32f9c/src/mavsdk/plugins/ftp/ftp_impl.cpp#L93-L95

```
_system_impl->call_user_callback(...);
```

Does that not work for your case?

Reply:
That works too. I changed it in https://github.com/mavlink/MAVSDK/pull/2272/commits/4297106d92780945363823b3e182c1e787b59d1a